pkg/controller/batch/batch_controller.go

```diff
@@ -0,0 +1,392 @@
+	// Ensure the Infinispan cluster exists
+	infinispan, err := r.loadCluster()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	batch := r.instance
+	spec := batch.Spec
+	if err := infinispan.EnsureClusterStability(); err != nil {
+		log.Info(fmt.Sprintf("Infinispan '%s' not ready: %s", spec.Cluster, err.Error()))
+		return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCluster}, nil
+	}
+
+	if spec.ConfigMap == nil && spec.Config == nil {
+		return reconcile.Result{},
+			r.UpdatePhase(v2.BatchFailed, fmt.Errorf("At most one of ['Spec.config', 'spec.ConfigMap'] must be configured"))
+	}
+
+	if spec.ConfigMap == nil {
+		// Create configMap
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      batch.Name,
+				Namespace: batch.Namespace,
+			},
+		}
+		_, err := controllerutil.CreateOrUpdate(ctx, r.client, configMap, func() error {
+			configMap.Data = map[string]string{BatchFilename: *spec.Config}
+			return controllerutil.SetControllerReference(batch, configMap, r.scheme)
+		})
+
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Unable to create ConfigMap '%s': %w", configMap.Name, err)
+		}
+
+		updated, err := r.update(func() {
+			batch.Spec.ConfigMap = &batch.Name
+		})
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Unable to update Batch .Spec: %w", err)
+		}
+		if updated {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// We update the phase separately to the spec as the status update is ignored when in the update mutate function
+	return reconcile.Result{}, r.UpdatePhase(v2.BatchInitialized, nil)
+}
+
+func (r *batchResource) execute() (reconcile.Result, error) {
+	infinispan, err := r.loadCluster()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	batch := r.instance
+	url, err := connectionUrl(infinispan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	cliArgs := fmt.Sprintf("--connect '%s' --file '%s/%s'", url, BatchVolumeRoot, BatchFilename)
+
+	labels := batchLabels(batch.Name)
+	infinispan.AddLabelsForPods(labels)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      batch.Name,
+			Namespace: batch.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32Ptr(0),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:    batch.Name,
+						Image:   infinispan.ImageName(),
+						Command: []string{Entrypoint, cliArgs},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      BatchVolumeName,
+								MountPath: BatchVolumeRoot,
+							},
+						},
+					}},
+					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes: []corev1.Volume{{
+						Name: BatchVolumeName,
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{Name: batch.Name},
```

Comment:
Is ConfigMap name always equals Batch CR name?

Reply:
Nope! Good spot. I have fixed this code and updated `TestBatchConfigMap` to specify a ConfigMap with a different name to the batch.